frameit/lib/frameit/device.rb

```diff
@@ -0,0 +1,102 @@
+require 'fastimage'
+require_relative 'module'
+
+module Frameit
+  class Device
+    attr_reader :id
+    attr_reader :deliver_screen_id
+    attr_reader :formatted_name
+    attr_reader :resolutions
+    attr_reader :density_ppi
+    attr_reader :default_color
+    attr_reader :platform
+    attr_reader :priority_config_key
+
+    def initialize(id, formatted_name, priority, resolutions, density_ppi, default_color, platform = Platform::IOS, deliver_screen_id = nil, priority_config_key = nil)
+      @id = id
+      @deliver_screen_id = deliver_screen_id
+      @formatted_name = formatted_name
+      @priority = priority
+      @resolutions = resolutions
+      @density_ppi = density_ppi
+      @default_color = default_color
+      @platform = platform
+      @priority_config_key = priority_config_key
+    end
+
+    def priority
+      if !priority_config_key.nil? && Frameit.config[priority_config_key]
+        999
+      else
+        @priority
+      end
+    end
+
+    def is_chosen_platform?(platform)
+      @platform == platform || platform == Platform::ANY
+    end
+
+    def formatted_name_without_apple
+      formatted_name.gsub("Apple", "").strip.to_s
+    end
+
+    def self.detect_device(path, platform)
+      size = FastImage.size(path)
+
+      UI.user_error!("Could not find or parse file at path '#{path}'") if size.nil? || size.count == 0
+
+      found_device = nil
+      filename_device = nil
+      filename = Pathname.new(path).basename.to_s
+      Devices.constants.each do |c|
+        device = Devices.const_get(c)
+        next unless device.resolutions.include?(size)
+        # assign to filename_device if the filename contains the formatted name / id and its priority is higher than the current filename_device
+        filename_device = device if (filename.include?(device.formatted_name_without_apple) || filename.include?(device.id)) && (filename_device.nil? || filename_device.priority < device.priority)
+        next unless device.is_chosen_platform?(platform) && (found_device.nil? || device.priority > found_device.priority)
+        found_device = device
+      end
+
+      # prefer filename
+      unless filename_device.nil?
+        return filename_device
+      end
```

Comment:
I agree, I made this code because I am used to Java where I usually do something like this:

`if (filenameDevice != null) return filenameDevice;`

And later changed it to the code above because of rubocop requiring using `.nil?` instead.

Reply:
[Fixed in the new commit.](https://github.com/milhauscz/fastlane/commit/d92a21b6c2cdce99009cd7f1b826ccc20c3a5fae)